### PR TITLE
feat: added realtime client side validation on checkout form inputs

### PR DESCRIPTION
**Pull request by Ankesh2004**
## Type of Change
- [ ] Enhancement

## Description
Adds realtime input validations on checkout page.

### Additional Changes
- Creates `checkout-validation.js` in js folder.
- Adds above script in `hyperswitch-checkout.php`

## Motivation and Context
Fixes issue juspay/hyperswitch#6169

## How did you test it?
Locally tested 
image

### CODE REVIEW

**Comment by sakksham7** [comment]
Hey @Ankesh2004 can you please add a screen recording of how validations will work covering all edge cases along with the screenshot you have provided.

**Comment by gorakhnathy7** [comment]
Hey @Ankesh2004 Any updates.

File: hyperswitch-checkout.php [hunk 1]
```diff
@@ -37,6 +37,19 @@
...
+        wp_enqueue_script(
+            'checkout-validation',
+            plugin_dir_url(__FILE__) . 'js/checkout-validation.js',
+            array('jquery'),
+            '1.0.0',
+            true
+        );
+    }
+}
+add_action('wp_enqueue_scripts', 'enqueue_checkout_validation_script');
```
Comment (vsrivatsa-edinburgh) [inline hyperswitch-checkout.php:51]:
```suggestion
add_action( 'wp_enqueue_scripts', 'enqueue_checkout_validation_script' );
```

File: js/checkout-validation.js [hunk 2]
```diff
@@ -0,0 +1,173 @@
...
+        console.error('Checkout form not found');
+        return;
+    }
+    console.log('Checkout form found');
+
+    if (!placeOrderButton) {
+        console.error('Place Order button not found');
+        return;
+    }
+    console.log('Place Order button found');
```
Comment (vsrivatsa-edinburgh) [inline js/checkout-validation.js:15]:
```suggestion
```

File: js/checkout-validation.js [hunk 3]
```diff
@@ -0,0 +1,173 @@
...
+    if (!form) {
+        console.error('Checkout form not found');
+        return;
+    }
+    console.log('Checkout form found');
+
+    if (!placeOrderButton) {
+        console.error('Place Order button not found');
+        return;
+    }
```
Comment (vsrivatsa-edinburgh) [inline js/checkout-validation.js:14]:
```suggestion
```

File: js/checkout-validation.js [hunk 4]
```diff
@@ -0,0 +1,173 @@
...
+    const addressField = form.querySelector('input[id="shipping-address_1"]');
+    const cityField = form.querySelector('input[id="shipping-city"]');
+    const stateField = form.querySelector('select[id="shipping-state"]');
+    const zipCodeField = form.querySelector('input[id="shipping-postcode"]');
+    const phoneField = form.querySelector('input[id="shipping-phone"]');
+
+    if (!emailField || !firstNameField || !lastNameField || !zipCodeField || !phoneField) {
+        console.error('One or more form fields not found');
+        return;
+    }
```
Comment (vsrivatsa-edinburgh) [inline js/checkout-validation.js:29]:
```suggestion
```

**Comment by vsrivatsa-edinburgh** [comment]
This is good work
I'm wondering how to solve this without using hardcoded error messages. Does WC expose any functions on their scripts that can be used to spit out respective error messages - that way you wouldn't have to worry about error messages in other languages
